onnxruntime/core/providers/cpu/tensor/gather_nd.cc

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "gather_nd.h"
+
+namespace onnxruntime {
+
+// Register a kernel for kMsDomain (contrib op) GatherND
+#ifndef DISABLE_CONTRIB_OPS
+
+namespace contrib {
+// TODO: Remove this contrib kernel registration and the schema from the appropriate places
+// once Keras Mask RCNN is shipped with all ONNX domain ops
+
+// Currently this kernel is required to support Keras Mask-RCNN
+ONNX_OPERATOR_KERNEL_EX(GatherND, kMSDomain, 1, kCpuExecutionProvider,
+                        KernelDefBuilder()
+                            .TypeConstraint("T", DataTypeImpl::AllTensorTypes())
+                            // contrib spec supports `int32_t` and `int64_t` for indices
+                            .TypeConstraint("Tind", {DataTypeImpl::GetTensorType<int32_t>(),
+                                                     DataTypeImpl::GetTensorType<int64_t>()}),
+                        GatherND);
+
+}  // namespace contrib
+
+#endif
+
+ONNX_CPU_OPERATOR_KERNEL(GatherND, 11,
+                         KernelDefBuilder()
+                             .TypeConstraint("T", DataTypeImpl::AllTensorTypes())
+                             // official ONNX spec only supports `int64_t` for indices
+                             .TypeConstraint("Tind", DataTypeImpl::GetTensorType<int64_t>()),
+                         GatherND);
+
+template <typename Tind>
+Status GatherNDBase::PrepareForCompute(OpKernelContext* context, Prepare& p) const {
+  auto input_tensor = context->Input<Tensor>(0);
+  auto indice_tensor = context->Input<Tensor>(1);
+  ORT_ENFORCE(input_tensor != nullptr);
+  ORT_ENFORCE(indice_tensor != nullptr);
+
+  auto input_shape = input_tensor->Shape();
+  auto indice_shape = indice_tensor->Shape();
```

Comment:
Most important is to use '&' with auto, as this is currently copying the vector.

Reply:
Fixed.